pytypes/typechecker.pyi

```diff
@@ -0,0 +1,48 @@
+# Generated using mypy stubgen, then editing to add types for some definitions.
+
+import __builtin__ as builtins
+from . import type_util, util
+from .exceptions import InputTypeError, OverrideError, ReturnTypeError
+from .type_util import TypeAgent, deep_type, get_orig_class, has_type_hints, is_Tuple, is_builtin_type, type_str
+from .util import getargspecs
+from types import ModuleType
+from typing import Any, Optional, Callable, TypeVar, Union
+from warnings import warn
+
+_python___import__ = builtins.__import__
+
+# Placeholders typed as Any just declare they exist to avoid "no such attribute"
+# errors from attempts to reference names missing from pyi.
+# TODO: Fill in proper types.
+def _pytypes___import__(name: Any, globls: Optional[Any] = ..., locls: Optional[Any] = ..., fromlist: Any = ..., level: int = ...): ...
+
+class _DelayedCheck:
+    func: Any = ...
+    method: Any = ...
+    class_name: Any = ...
+    base_method: Any = ...
+    base_class: Any = ...
+    exc_info: Any = ...
+    raising_module_name: Any = ...
+    def __init__(self, func: Any, method: Any, class_name: Any, base_method: Any, base_class: Any, exc_info: Any) -> None: ...
+    def run_check(self, raise_NameError: bool = ...) -> None: ...
+
+T_Callable = TypeVar('T_Callable', bound=Callable)
+def override(func: T_Callable, auto: bool = ...) -> T_Callable: ...
+def typechecked_func(func: T_Callable, force: bool = ..., argType: Optional[Any] = ..., resType: Optional[Any] = ..., prop_getter: bool = ...): ...
+def typechecked_class(cls, force: bool = ..., force_recursive: bool = ...): ...
+def typechecked_module(md: Any, force_recursive: bool = ...): ...
```

Comment:
Shouldn't ``md: Any`` rather be ``md: Union[ModuleType, str]``? (same for ``auto_override_module`` below)

Reply:
Or even ``T_Mod = TypeVar('T_Mod', bound=Union[ModuleType, str])`` and then ``def typechecked_module(md: T_Mod, force_recursive: bool = ...) -> T_Mod: ...`` ?

Reply:
K, added. I just had trouble determining some of these exact expected types from reading the code and hoped not to keep adding a little more and a little more to the scope of this initial change.